pkgs/by-name/tr/treecat/package.nix

```diff
@@ -0,0 +1,99 @@
+{ stdenv
+, fetchFromSourcehut
+, hare
+, harec
+, lib
+, scdoc
+}:
+let
+  # treecat needs a more recent version of Hare. Until #265124 is merged onto
+  # master (it's already merged on staging), these overrides are needed.
+  harec' = harec.overrideAttrs (_old: {
+    version = "unstable-2023-11-28";
+    src = fetchFromSourcehut {
+      owner = "~sircmpwn";
+      repo = "harec";
+      rev = "ec3193e3870436180b0f3df82b769adc57a1c099";
+      hash = "sha256-HXQIgFC4YVDJjo5xbyg1ea3jWYKLEwKkD1KFzWFz9UI=";
+    };
+  });
+  hare' = (hare.overrideAttrs (_old: {
+    version = "unstable-2023-11-27";
+    src = fetchFromSourcehut {
+      owner = "~sircmpwn";
+      repo = "hare";
+      rev = "d94f355481a320fb2aec13ef62cb3bfe2416f5e4";
+      hash = "sha256-Mpl3VO4xvLCKHeYr/FPuS6jl8CkyeqDz18mQ6Zv05oc=";
+    };
+    # checkPhase needs the tzdata file added on the aforementioned PR.
+    doCheck = false;
+  })).override { harec = harec'; };
```

Comment:
Rather than pin harec it would make sense to either wait or target staging(-next)

Reply:
One could use `staging` or `staging-next` if the change needed by one's PR was already on either one those. Is this understanding correct?

Reply:
> > is merged onto master (it's already merged on staging)
> 
> I assumed it was merged onto staging. Sometimes PR that depends on functionality currently in staging target staging to become a part of the staging->master merge

I can see the confusion: What was actually merged onto `staging` --- and later on `staging-next` was the [`tzdata` PR](https://github.com/NixOS/nixpkgs/pull/265124). But I could have set the [`hare`'s PR](https://github.com/NixOS/nixpkgs/pull/271476) base branch as `staging-next`, right?

Actually, since it is still possible to change the base branch on the latter, would it be a good idea to change it to `staging-next`? The only reason [#271476](https://github.com/NixOS/nixpkgs/pull/271476) is a draft is because it is waiting for [#265124](https://github.com/NixOS/nixpkgs/pull/265124) to be merged onto `master`.